routers/web/auth/oauth2_provider.go

```diff
@@ -104,7 +104,15 @@ func InfoOAuth(ctx *context.Context) {
 		Picture:           ctx.Doer.AvatarLink(ctx),
 	}
 
-	groups, err := oauth2_provider.GetOAuthGroupsForUser(ctx, ctx.Doer)
+	var accessTokenScope auth.AccessTokenScope
+	if auHead := ctx.Req.Header.Get("Authorization"); auHead != "" {
+		auths := strings.Fields(auHead)
+		if len(auths) == 2 && (auths[0] == "token" || strings.ToLower(auths[0]) == "bearer") {
+			accessTokenScope, _ = auth_service.GetOAuthAccessTokenScopeAndUserID(ctx, auths[1])
+		}
+	}
+	onlyPublicGroups, _ := accessTokenScope.PublicOnly()
```

Comment:
it is not an error `GetOAuthAccessTokenScopeAndUserID` returns `AccessTokenScope` and  `int64` (user ID). here we need accessTokenScope to be checked if PublicOnly and we don't need user ID so that's why omitted.

Reply:
But it is an error

![image](https://github.com/user-attachments/assets/383a8c2c-c61e-4ebd-9a12-0a85efa59f4d)

Reply:
aha. sorry my bad.

Reply:
so, here we are interested only if there's `public-only` so that `onlyPublicGroups` becomes `true`. anything else fallback to `false` and for the result it behaves as it is now: it does `IncludePrivate` groups. i don't see the benefit of handling an error. is that ok?

Reply:
Could it have some comments? Otherwise it's difficult to understand.